Refactor Main argument handling

KoBeWi opened the pull request:
This PR improves how launch options are registered and handled.
- Added LaunchOptions struct
  - Contains Option struct, which is an info about a single recognized launch option
  - Contains enum for all default launch options
- Each launch Option is properly registered with name, alises, description etc.
- Help is no longer hard-coded, it's printed based on the registered options
- Argument matching is performed based on registered options
- Config variables were moved from top level to dedicated structs

The main improvement is that all launch options are now registered in one place and handled in one place, instead of being spread all over the file. Makes them very easy to find and manage.

In a future PRs I plan to move some options out of main and register them in the modules. It could work similar to context menu plugins, i.e. you register an option, it gets an ID and associated callback that gets called.
Also more config options can be moved to structs (I didn't move all of them).

Note that registering options is a big cumbersome right now. There are many parameters and many default arguments get repeated. It could be improved with the new struct initialization in C++20, which basically works like named arguments. I added a sample line and constructor how it would look (unused for now).

KoBeWi committed:
Refactor Main argument handling

akien-mga commented:
GCC and Clang don't seem happy: https://github.com/godotengine/godot/actions/runs/15535896183/job/43734802381?pr=107209

KoBeWi commented:
Rebased and also changed the struct definition a bit, because I'm also affected by the gcc bug now -_-
Something broke last time I tested, so I'll try to update soon.
